examples/yarn-pnp/package.json

```
@@ -0,0 +1,33 @@
+{
+  "name": "remix-example-yarn-pnp",
+  "private": true,
+  "description": "",
+  "license": "",
+  "sideEffects": false,
+  "scripts": {
+    "build": "remix build",
+    "dev": "remix dev",
+    "start": "remix-serve build"
+  },
+  "dependencies": {
+    "@remix-run/node": "1.5.1",
+    "@remix-run/react": "1.5.1",
+    "@remix-run/serve": "1.5.1",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "~": "link:./app"
+  },
+  "devDependencies": {
+    "@remix-run/dev": "1.5.1",
+    "@types/react": "^17.0.24",
+    "@types/react-dom": "^17.0.9",
+    "@yarnpkg/esbuild-plugin-pnp": "^2.0.0",
+    "esbuild": ">=0.10.0",
+    "express": "^4.17.1",
+    "typescript": "^4.1.2"
+  },
+  "engines": {
+    "node": ">=14"
+  },
+  "packageManager": "yarn@3.2.1"
```

Let's remove the `packageManager` field & add `yarn` to the `engines` field instead, as Corepack is still experimental and we don't have a way to specify a range

```suggestion
  "engines": {
    "node": ">=14"
    "yarn": ">=3.0.0"
  }
```

---

I'm not sure. The recommended way to install yarn [_is_ using corepack](https://yarnpkg.com/getting-started/install), so I think it'd be better to keep the `"packageManager"` field as is.
At least on my Linux distribution, you only get yarn 1.x from the official repositories. To use yarn ≥ 2 (and PnP), you only really have two options:
- use corepack and the `"packageManager"` field
- run `yarn set version berry`, which copies the latest stable version of yarn into the project's `.yarn/releases` directory and put a reference to it in `.yarnrc.yml`. This local copy _must not be gitignored_ or you won't be able to run yarn in the project at all.

Doing the latter would work, but would add a `.cjs` file with ~2.1MiB, which I think is a bit much for such an (otherwise small) example project.